Use Vega-Embed

domoritz opened the pull request:
Fixes #1, fixes #8, fixes #7 

I'd like to use clean module imports as @mbostock described in https://github.com/observablehq/vega/issues/8#issuecomment-378389907 but this requires a few changes to our libraries that I will need to make later.

Includes style suggestions by @jheer. 

Here is a demo: https://beta.observablehq.com/@domoritz/vega-embed-observable-package-experiments

domoritz committed:
Use Vega-Embed. Fixes #1, fixes #8, fixes #7

domoritz commented:
Any feedback @mbostock and @tmcw?

tmcw commented:
I'm admittedly a little new to Vega: is the main advantage of vega-embed over our current setup in this situation the action links? It looks like the other billed ability is loading a visualization from URLs, which doesn't come into play in the Observable context, so I think that's it?

(assuming that's the case) I'm not certain - the action for viewing the the chart source is sort of redundant when the source code is readily available, and I'm not really sure that using the editor or exporting to PNG are so useful that they make sense as default UI elements. Maybe in certain cases they're worthwhile, but I think the minority, while making them default will mean that the data-actions UI will still take up y-space and perhaps distract from the visualization.

I'd personally err on the side of vega-embed being an option offered alongside, rather than in place of, the current `@observablehq/vega` module.

domoritz commented:
Thank you for the feedback. I agree that the code option is redundant and that the other options may not be needed by default. However, I found opening a spec in the editor useful. For example, you can print specs as PDF and use autocompete. In the near future, we will add more IDE functionalities to the editor that won't be supported in observable. 

Vega embed offers a few more options for example to change the renderer. I think it makes sense to give users access to these options. 

Besides embed, this PR gives users access to the Vega view.

domoritz committed:
Hide “show source” action

domoritz commented:
I've updated the code to remove the "show source" action.

domoritz committed:
Don’t use bundle, hide actions unless the user hovers over the Vega chart, attach style only once

domoritz commented:
As @mbostock suggested in https://github.com/observablehq/vega/issues/8#issuecomment-378389907, I switched Vega-Embed over to rollup. Vega-Embed is not compatible with d3-require and I updated this PR. I think it's good to go now. I updated https://beta.observablehq.com/@domoritz/vega-embed-observable-package-experiments so you can see it in action over there.

domoritz commented:
Any feedback @mbostock and @tmcw?

domoritz commented:
Vega-Embed now supports Observable out of the box: https://beta.observablehq.com/@domoritz/hello-vega-embed

🎉
